com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingAmbientOcclusion.cs

```diff
@@ -59,103 +54,279 @@ static RTHandle AmbientOcclusionHistoryBufferAllocatorFunction(string viewName,
 
                 // Run the computation
-                cmd.DispatchRays(aoShaderRT, m_RayGenShaderName, (uint)hdCamera.actualWidth, (uint)hdCamera.actualHeight, (uint)hdCamera.viewCount);
+                cmd.DispatchRays(aoTraceParameters.aoShaderRT, m_RayGenShaderName, (uint)aoTraceParameters.actualWidth, (uint)aoTraceParameters.actualHeight, (uint)aoTraceParameters.viewCount);
             }
+        }
+
+        public void DenoiseAO(CommandBuffer cmd, HDCamera hdCamera, AmbientOcclusionDenoiseParameters aoDenoiseParameters, AmbientOcclusionDenoiseResources aoDenoiseResources)
+        {
+            // If the ray tracing state is invalid, we should be doing anything
+            if (!aoDenoiseParameters.validRayTracing)
+                return;
 
             using (new ProfilingScope(cmd, ProfilingSampler.Get(HDProfileId.RaytracingFilterAmbientOcclusion)))
             {
-                if(aoSettings.denoise)
+                if (aoDenoiseParameters.denoise)
                 {
-                    // Grab the history buffer
-                    RTHandle ambientOcclusionHistory = hdCamera.GetCurrentFrameRT((int)HDCameraFrameHistoryType.RaytracedAmbientOcclusion)
-                        ?? hdCamera.AllocHistoryFrameRT((int)HDCameraFrameHistoryType.RaytracedAmbientOcclusion, AmbientOcclusionHistoryBufferAllocatorFunction, 1);
-
-                        float historyValidity = 1.0f;
-                    #if UNITY_HDRP_DXR_TESTS_DEFINE
-                        if (Application.isPlaying)
-                            historyValidity = 0.0f;
-                        else
-                    #endif
-                            // We need to check if something invalidated the history buffers
-                            historyValidity = m_RenderPipeline.ValidRayTracingHistory(hdCamera) ? 1.0f : 0.0f;
-
                     // Apply the temporal denoiser
                     HDTemporalFilter temporalFilter = m_RenderPipeline.GetTemporalFilter();
-                    temporalFilter.DenoiseBuffer(cmd, hdCamera, m_AOIntermediateBuffer0, ambientOcclusionHistory, m_AOIntermediateBuffer1, historyValidity: historyValidity);
+                    temporalFilter.DenoiseBuffer(cmd, hdCamera, aoDenoiseResources.inputTexture, aoDenoiseResources.ambientOcclusionHistory, aoDenoiseResources.intermediateBuffer, historyValidity: aoDenoiseParameters.historyValidity);
 
                     // Apply the diffuse denoiser
                     HDDiffuseDenoiser diffuseDenoiser = m_RenderPipeline.GetDiffuseDenoiser();
-                    diffuseDenoiser.DenoiseBuffer(cmd, hdCamera, m_AOIntermediateBuffer1, outputTexture, aoSettings.denoiserRadius);
+                    diffuseDenoiser.DenoiseBuffer(cmd, hdCamera, aoDenoiseResources.intermediateBuffer, aoDenoiseResources.outputTexture, aoDenoiseParameters.denoiserRadius);
                 }
                 else
```

Comment:
Why the blit here? Is it just because the input texture might be reused somewhere else? Are they the same format?
With RenderGraph we might not need this kind of patterns, so basically the denoise pass would just return the input if no denoising is needed (which would be outside of the static function in this case).
In general to better map to render graph, the static functions that needs to do actual rendering should not have different path and should only do one thing. This way they can better share resources for example.